can-ajax-test.js

```diff
@@ -607,3 +611,117 @@ QUnit.test("beforsend", function (assert) {
 		done();
 	});
 });
+
+QUnit.test("abort", function (assert) {
+	var done = assert.async();
+	var restore = makeFixture(function () {
+		var aborted = false;
+	    this.open = function (type, url) {};
+	    this.setRequestHeader = function (header, value) {};
+	    this.send = function () {};
+	    this.abort = function() {
+	    	assert.ok(true, 'called the underlying XHR.abort');
+	    	done();
+	    };
+    });
+
+	var request = ajax({
+		url: "/foo"
+	});
+
+	request.abort();
+});
+
+QUnit.test("abort prevents sending if beforeSend is not finished", function (assert) {
+	var done = assert.async();
+	var restore = makeFixture(function () {
+		var aborted = false;
+	    this.open = function (type, url) {};
+	    this.setRequestHeader = function (header, value) {};
+	    this.abort = function() {
+	    	assert.ok(true, 'XHR abort was called');
+	    };
+	    this.send = function () {
+	    	assert.notOk(true, 'should not have been called');
+	    };
+    });
+
+	var request = ajax({
+		url: "/foo",
+		beforeSend: function (xhr){
+			return new Promise(resolve => {
+				setTimeout(resolve, 1);
+			});
+		}
+	});
+
+	request.abort();
+
+	setTimeout(done, 10);
+});
+
+QUnit.module("beforeSend async", function(hooks) {
```

Comment:
will this include `can-ajax` in the test name?  I don't think it will

Reply:
Fixed. I did not need the hooks anyways, so I just removed the module wrapper altogether.